src/Akka.Streams.Kafka/Stages/Consumers/Abstract/SingleSourceStageLogic.cs

```diff
@@ -22,67 +22,53 @@ internal class SingleSourceStageLogic<K, V, TMessage> : BaseSingleSourceLogic<K,
     {
         private readonly SourceShape<TMessage> _shape;
         private readonly ConsumerSettings<K, V> _settings;
-        private readonly ISubscription _subscription;
 
         public SingleSourceStageLogic(SourceShape<TMessage> shape, ConsumerSettings<K, V> settings, 
                                       ISubscription subscription, Attributes attributes, 
                                       Func<BaseSingleSourceLogic<K, V, TMessage>, IMessageBuilder<K, V, TMessage>> messageBuilderFactory) 
-            : base(shape, attributes, messageBuilderFactory, settings.AutoCreateTopicsEnabled)
+            : base(shape, attributes, messageBuilderFactory, settings.AutoCreateTopicsEnabled, subscription)
         {
             _shape = shape;
             _settings = settings;
-            _subscription = subscription;
         }
 
-        /// <inheritdoc />
-        protected override void ConfigureSubscription()
+        protected override object LogSource
         {
-            switch (_subscription)
+            get
             {
-                case TopicSubscription topicSubscription:
-                    ConsumerActor.Tell(new KafkaConsumerActorMetadata.Internal.Subscribe(topicSubscription.Topics), SourceActor.Ref);
-                    break;
-                case TopicSubscriptionPattern topicSubscriptionPattern:
-                    ConsumerActor.Tell(new KafkaConsumerActorMetadata.Internal.SubscribePattern(topicSubscriptionPattern.TopicPattern), SourceActor.Ref);
-                    break;
-                case IManualSubscription manualSubscription:
-                    ConfigureManualSubscription(manualSubscription);
-                    break;
-                default:
-                    throw new NotSupportedException();
+                var strPart = (_settings.Properties.TryGetValue("client.id", out var clientId)) ?
+                    $"client-{_settings.GroupId}-{clientId}" : $"client-{_settings.GroupId}";
+                return Akka.Event.LogSource.Create(strPart, GetType());
             }
         }
 
         /// <inheritdoc />
         protected override IActorRef CreateConsumerActor()
         {
-            var partitionsAssignedHandler = GetAsyncCallback<IEnumerable<TopicPartition>>(PartitionsAssigned);
```

Comment:
All of the event handler configuration code got moved to `ConfigureSubscription`, which happens after the `KafkaConsumerActor` is created.